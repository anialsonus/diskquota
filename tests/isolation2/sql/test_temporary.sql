1: CREATE SCHEMA temporary_schema;
1: SET search_path TO temporary_schema;
1: SELECT diskquota.set_schema_quota('temporary_schema', '1 MB');
1: SELECT diskquota.wait_for_worker_new_epoch();
1: CREATE TEMPORARY TABLE temporary_table(id int) DISTRIBUTED BY (id);
1: INSERT INTO temporary_table SELECT generate_series(1, 100000);
1: SELECT diskquota.wait_for_worker_new_epoch();
1q:

!\retcode gpstop -afr;

1: SELECT pg_table_size('temporary_table'::regclass::oid) size;
1: SELECT size FROM diskquota.table_size WHERE NOT EXISTS (SELECT 1 FROM pg_class WHERE tableid = oid) AND segid = -1;
1: DROP SCHEMA temporary_schema CASCADE;
1q:
