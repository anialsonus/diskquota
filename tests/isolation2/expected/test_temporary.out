1: CREATE SCHEMA temporary_schema;
CREATE
1: SET search_path TO temporary_schema;
SET
1: SELECT diskquota.set_schema_quota('temporary_schema', '1 MB');
 set_schema_quota 
------------------
                  
(1 row)
1: SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t                         
(1 row)
1: CREATE TEMPORARY TABLE temporary_table(id int) DISTRIBUTED BY (id);
CREATE
1: INSERT INTO temporary_table SELECT generate_series(1, 100000);
INSERT 100000
1: SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t                         
(1 row)
1q: ... <quitting>

!\retcode gpstop -afr;
-- start_ignore
-- end_ignore
(exited with code 0)

1: SELECT pg_table_size('temporary_table'::regclass::oid) size;
ERROR:  relation "temporary_table" does not exist
LINE 1: SELECT pg_table_size('temporary_table'::regclass::oid) size;
                             ^
1: SELECT size FROM diskquota.table_size WHERE NOT EXISTS (SELECT 1 FROM pg_class WHERE tableid = oid) AND segid = -1;
 size 
------
(0 rows)
1: DROP SCHEMA temporary_schema CASCADE;
DROP
1q: ... <quitting>
